lib/rubocop/cop/mixin/space_before_punctuation.rb

```diff
@@ -10,7 +10,7 @@ module SpaceBeforePunctuation
       def investigate(processed_source)
         processed_source.tokens.each_cons(2) do |t1, t2|
           next unless kind(t2) && t1.pos.line == t2.pos.line &&
-                      t2.pos.begin_pos > t1.pos.end_pos
+                      t2.pos.begin_pos > t1.pos.end_pos && t1.type != :tLCURLY
```

Comment:
I think we should skip the `:tLCURLY` only if `Style/SpaceInsideBlockBraces` is enabled and its `EnforcedStyle` is `space`. It's not as simple as this solution, but it adds clarity about the reason. Also, if the `SpaceInsideBlockBraces` style is `no_space`, then `test { ; }` should be auto-corrected to `test {;}`.